library/ziti.c

```diff
@@ -882,9 +990,25 @@ static void edge_routers_cb(ziti_edge_router_array ers, const ziti_error *err, v
     }
 }
 
-static void session_post_auth_query_cb(ziti_context ztx, int status, void *ctx){
+static void update_identity_data(ziti_identity_data *data, const ziti_error *err, void *ctx) {
+    ziti_context ztx = ctx;
+
+    if (err) {
+        ZITI_LOG(ERROR, "failed to get identity_data: %s[%d]", err->message, err->code);
+    } else {
+        free_ziti_identity_data(ztx->identity_data);
+        FREE(ztx->identity_data);
+        ztx->identity_data = data;
+    }
+
+    update_ctrl_status(ztx, FIELD_OR_ELSE(err, err, 0), FIELD_OR_NULL(err, message));
+}
+
+static void session_post_auth_query_cb(ziti_context ztx, int status, void *ctx) {
     ziti_session *session = ztx->session;
 
+    ziti_ctrl_current_identity(&ztx->controller, update_identity_data, ztx);
```

Comment:
This was moved to after MFA is completed. It was 401'ing all the time on MFA identities.